src/station_cmd.cpp

```diff
@@ -2162,20 +2162,32 @@ uint8 GetAirportNoiseLevelForDistance(const AirportSpec *as, uint distance)
  */
 Town *AirportGetNearestTown(const AirportSpec *as, const TileIterator &it, uint &mindist)
 {
-	Town *t, *nearest = nullptr;
-	uint add = as->size_x + as->size_y - 2; // GetMinimalAirportDistanceToTile can differ from DistanceManhattan by this much
-	mindist = UINT_MAX - add; // prevent overflow
-	FOR_ALL_TOWNS(t) {
-		if (DistanceManhattan(t->xy, it) < mindist + add) { // avoid calling GetMinimalAirportDistanceToTile too often
-			TileIterator *copy = it.Clone();
-			uint dist = GetMinimalAirportDistanceToTile(*copy, t->xy);
-			delete copy;
+	assert(Town::GetNumItems() > 0);
+
+	Town *nearest = nullptr;
+
+	uint it_x = TileX(it);
+	uint it_y = TileY(it);
+	uint max_it_x = it_x + as->size_x - 1;
+	uint max_it_y = it_y + as->size_y - 1;
+
+	mindist = UINT_MAX - 1; // prevent overflow
```

Comment:
Again, comment why you need the you need this. (If two airports are the same distance, you want the one with the lowest index)

Reply:
It is in the description.